src/wallet.js

```diff
@@ -385,6 +385,22 @@ const wallet = {
     return promise;
   },
 
+  /**
+   * Update the last generated address index with a new candidate index if it is
+   * greater than the current last generated index.
+   *
+   * @param {number} newIndex The candidate index for the last generated index
+   * @returns {boolean} Whether the last generated index was updated
+   */
+  updateLastGeneratedIndex(newIndex) {
```

Comment:
Not sure I like the method name because it may not update the last generated index.

I couldn't think of a good name, to be honest, maybe `conditionallyUpdateLastGeneratedIndex`, or `updateLastGeneratedIndexIfGreater`. Not sure the best approach but the current name is a bit confusing in my opinion.